### PR TITLE
[MIG] stock_by_warehouse: Migration to 14.0

### PR DESCRIPTION
**Pull request by hugho-ad**

### CODE REVIEW

**Comment by CarmenMiranda** [comment]
Hi @hugho-ad It looks good to me, however, I'm not completely familiar with `@api.depends_context` (I'm still working with v12) so if there is someone else that is more familiar with them I would suggest you check it out

Something that I saw is that the following commits from 12 (which are some improvements) are missing :

- [[ADD] stock_by_warehouse: add context for multi-warehouse](https://github.com/Vauxoo/addons-vauxoo/commit/f1e72550f03b343841a0f2590442e089724b8f5a#diff-d0ef3ea8e1fc0597cd430219f8ee2d0da02cfd730b585c626cd2949d32e91f93)
- [[REF] stock_by_warehouse: Locations Widget](https://github.com/Vauxoo/addons-vauxoo/commit/c2fe1362f1980f576dbe1d445f019583378668a4#diff-d0ef3ea8e1fc0597cd430219f8ee2d0da02cfd730b585c626cd2949d32e91f93)
- [[REF] stock_by_warehouse: Product's Context (#1490)](https://github.com/Vauxoo/addons-vauxoo/commit/67255ed14026a18dc126c8aa19c5781007b7c1ec#diff-d0ef3ea8e1fc0597cd430219f8ee2d0da02cfd730b585c626cd2949d32e91f93)

**Comment by hugho-ad** [comment]
@CarmenMiranda 

I'm migrating from v13.0, that is why the improvement commits you mention are not here

I could made the forward port in a separated MR

about depends_context that changes were made when we migrate the module to v13.0, I already request @deivislaya reviewing.

**Comment by CarmenMiranda** [comment]
> @CarmenMiranda
> 
> I'm migrating from v13.0, that is why the improvement commits you mention are not here
> 
> I could made the forward port in a separated MR
> 
> about depends_context that changes were made when we migrate the module to v13.0, I already request @deivislaya reviewing.

Oki, besides that LGTM 👍🏼

**Comment by hugho-ad** [comment]
@moylop260 could you review pls

**Comment by moylop260** [comment]
I think odoo v14.0 doesn't need all the workaround that I used here for v12.0

But even it is working so
👍

File: stock_by_warehouse/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+    def _prepare_domain_available_not_reserved(self):
+        domain_quant = [
+            ('product_id', 'in', self.ids)
+        ]
+        domain_quant_locations = self._get_domain_locations()[0]
+        domain_quant.extend(domain_quant_locations)
+        return domain_quant
+
+    @api.depends_context('warehouse', 'force_company')
```
Comment (deivislaya) [inline stock_by_warehouse/models/product_product.py]: Delete  ``` force_company ``` https://github.com/odoo/odoo/blob/14.0/odoo/models.py#L5100

File: stock_by_warehouse/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                'saleable':
+                product.qty_available - product.outgoing_qty
+            })
+        return json.dumps(info)
+
+    @api.depends_context(
+        'lot_id', 'owner_id', 'package_id', 'from_date', 'to_date',
+        'company_owned', 'force_company', 'warehouse',
+    )
+    def _compute_quantities(self):
```
Comment (deivislaya) [inline stock_by_warehouse/models/product_product.py]: @hugho-ad 

This inheritance is no longer necessary, the missing field "warehouse" was fixed in this commit  https://github.com/odoo/odoo/commit/d6bd78e578104f43318fbe3c11f325063e3e5042  and V14.0 it is also fixed https://github.com/odoo/odoo/blob/14.0/addons/stock/models/product.py#L100 

Regards !

File: stock_by_warehouse/models/product_template.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                'virtual': tmpl.virtual_available,
+                'incoming': tmpl.incoming_qty,
+                'outgoing': tmpl.outgoing_qty,
+                'saleable':
+                tmpl.qty_available - tmpl.outgoing_qty
+            })
+        return json.dumps(info)
+
+    @api.depends_context('company_owned', 'force_company', 'warehouse')
+    def _compute_quantities(self):
```
Comment (deivislaya) [inline stock_by_warehouse/models/product_template.py]: @hugho-ad  Also here

This inheritance is no longer necessary, the missing field "warehouse" was fixed in this commit  https://github.com/odoo/odoo/commit/d6bd78e578104f43318fbe3c11f325063e3e5042 

Regards !

File: stock_by_warehouse/models/product_product.py (hunk 1 above)
Comment (hugho-ad) [reply stock_by_warehouse/models/product_product.py]: thanks!, I replace it with company
https://github.com/odoo/odoo/blob/14.0/odoo/api.py#L226

File: stock_by_warehouse/models/product_template.py (hunk 3 above)
Comment (hugho-ad) [reply stock_by_warehouse/models/product_template.py]: thanks!

done!

**Comment by deivislaya** [comment]
LGTM  :+1: 

Good job!!

Regards

**Comment by luisg123v** [comment]
@hugho-ad remember to remove branches after merging.

**Pull request merged**